enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,445 @@
+   additional interfaces based on matching NetworkAttachmentDefinitions
+1. Application's containers are running, they can utilize additional interfaces
+
+### API Extensions
+
+Multus is an established project with already existing API extensions.
+Following paragraphs does not present brand new CRDs or APIs, it only aims to summarize how adding
+Multus will affect MicroShift's API. For more information see
+[Multus CNI repository](https://github.com/openshift/multus-cni/).
+
+Multus is configured in following ways:
+- CNI configuration created when Multus' DaemonSet starts, it includes the primary CNI (ovn-kubernetes for MicroShift)
+  that is default delegate (CNI invoked for all Pods). It can be crafted manually but usually it's
+  autogenerated based on current primary CNI and options provided to the script.
+- NetworkAttachmentDefinition (net-attach-def) CR
+- Annotations
+
+NetworkAttachmentDefinition (NAD) is a simple CR which only contains single string field called `config` which
+contains a JSON-formatted CNI configuration.
+Exact schema of the CNI config depends on the specific CNI and its documentation must be consulted.
+Exact values depend on the runtime environment as configs may require name of the specific host interface.
+For these reasons, we can only provide examples and guidance on crafting CNI configs, therefore we
+cannot package a default configuration that could work on every host in `microshift-networking-multus` RPM.
+
+Below is example of NAD to use the `bridge` CNI with host bridge named `test-bridge`:
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: bridge-conf
+spec:         
+  config: '{                                
+      "cniVersion": "0.3.0",
+      "type": "bridge",
+      "bridge": "test-bridge",
+      "mode": "bridge",                                                   
+      "ipam": {  
+        "type": "host-local",
+        "subnet": "192.168.20.0/24",
+        "rangeStart": "192.168.20.200",
+        "rangeEnd": "192.168.20.216",
+        "routes": [
+          { "dst": "0.0.0.0/0" }
+        ],
+        "gateway": "192.168.20.1"
+      }
+    }'
+```
+
+There are two Pod annotation that can be used to instruct Multus on how the networks should be set up.
+Main one is `k8s.v1.cni.cncf.io/networks` which specifies which NAD should be added to the Pod.
+Multiple NADs can be specified by separating them with comma. NADs can be even reused.
+Examples from Multus docs:
+```yaml
+apiVersion: v1                  
+kind: Pod                        
+metadata:
+  name: samplepod  
+  annotations:
+    k8s.v1.cni.cncf.io/networks: bridge-conf
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: samplepod
+  annotations:
+    k8s.v1.cni.cncf.io/networks: bridge-conf,bridge-conf
+```
+
+Other Pod annotation is `v1.multus-cni.io/default-network` which overrides default CNI from Multus'
```

Comment:
Need to double check with @pliurh, the original author of `default-network` annotation, but as far as I know, this default-network annotation is desgined only for ovn migration and it is not in multus document as well. It might be better not to mention that in the document if there is no specific use-cases for microshift.

Reply:
removed mention of it